### PR TITLE
Utopia 847 - Next Steps

### PR DESCRIPTION
**Pull request by antsand**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/public/PIAFormTabs/intake/index.tsx [hunk 1]
```diff
@@ -35,6 +35,19 @@ export const PIAFormIntake = () => {
...
+      return "I'm not sure";
+    }
+    if (pia?.hasAddedPiToDataElements === true) {
+      return 'Yes';
+    }
+    if (pia?.hasAddedPiToDataElements === false) {
+      return 'No';
+    }
+    return '';
+  };
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/intake/index.tsx:49]: can we please leverage "PIOptions" here? No independent string comparisons

File: src/frontend/src/components/public/PIASubHeader/index.tsx [hunk 2]
```diff
@@ -24,6 +25,9 @@ function PIASubHeader({
   // this pr just add download function for faEllipsisH icon
   const [isDownloading, setIsDownloading] = useState(false);
   const [downloadError, setDownloadError] = useState('');
+  const { pathname } = useLocation();
+  const nextStepAction = pathname?.split('/').includes('nextSteps');
+  console.log(nextStepAction);
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIASubHeader/index.tsx:30]: console log

File: src/frontend/src/pages/PIAIntakeForm/index.tsx [hunk 3]
```diff
@@ -212,9 +214,18 @@ const PIAFormPage = () => {
...
+          navigate(
+            buildDynamicPath(routes.PIA_NEXT_STEPS_EDIT, {
+              id: pia.id,
+              title: pia.title,
+            }),
+          );
+        } else {
+          navigate(routes.PIA_NEXT_STEPS_EDIT, {
+            state: { result: updatedPia },
+          });
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAIntakeForm/index.tsx:227]: if pia.id is not there, we should not perform any action.

Maybe disable the submit button, until we have pia.id ??

File: src/frontend/src/sass/_nextSteps.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.nextSteps {
+    .button-container {
+        :first-child {
+            margin-right: 1em;
+        }
+    }
+}
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_nextSteps.scss:7]: missing EOL

**Comment by kushal-arora-fw** [comment]
Local Testing results:
- Submit button - is changing status to MPO Review
- Submitting leads straight to the first tab or view page based on PI [No Next Steps page is shown] [--useEffect not working as we expected] - I will look into this piece.

File: src/frontend/src/components/public/PIAFormTabs/intake/index.tsx (hunk 1 above)
Comment (antsand) [reply src/frontend/src/components/public/PIAFormTabs/intake/index.tsx:49]: somewhat resolved

File: src/frontend/src/pages/PIAIntakeForm/index.tsx (hunk 3 above)
Comment (antsand) [reply src/frontend/src/pages/PIAIntakeForm/index.tsx:227]: needs to be addresses more broadly.

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**